### PR TITLE
[FIX] *: Adapt old custom-control class of Bootstrap

### PR DESCRIPTION
**Pull request by abz-odoo**
* = hr_holidays, mail, sale, web, web_editor, website, website_forum,
website_payment, website_sale

https://getbootstrap.com/docs/5.1/migration/

replacing `custom-control` by `form-check` because in bs4 this class had
a use which is not the case in bs5.

### CODE REVIEW

**Comment by abz-odoo** [comment]
@rfr-odoo
Hello ! It is okay to you ? :)

File: addons/hr_holidays/static/src/radio_image_field/radio_image_field.xml [hunk 1]
```diff
@@ -12,7 +12,7 @@
             </t>
             <t t-else="">
                 <t t-foreach="items" t-as="item" t-key="item[0]">
-                    <div class="custom-control form-check o_radio_item" aria-atomic="true">
+                    <div class="form-check form-check o_radio_item" aria-atomic="true">
```
Comment (rfr-odoo) [inline addons/hr_holidays/static/src/radio_image_field/radio_image_field.xml:15]: There are some duplicate left (`form-check form-check` in this case) in the code due to the replacement.

File: addons/web/static/src/legacy/scss/bootstrap_overridden.scss [hunk 2]
```diff
@@ -168,15 +168,15 @@ $list-group-active-color: $o-list-group-active-color !default;
...
-$custom-control-indicator-checked-bg: $primary !default;
-$custom-control-indicator-checked-border-color: $primary !default;
+$form-check-indicator-checked-color: $white !default;
+$form-check-indicator-checked-bg: $primary !default;
+$form-check-indicator-checked-border-color: $primary !default;
 
-$custom-control-indicator-disabled-bg: $gray-200 !default;
-$custom-control-indicator-checked-disabled-bg: $gray-200 !default;
+$form-check-indicator-disabled-bg: $gray-200 !default;
+$form-check-indicator-checked-disabled-bg: $gray-200 !default;
```
Comment (rfr-odoo) [inline addons/web/static/src/legacy/scss/bootstrap_overridden.scss:179]: It seems that old variable don't exist anymore in Bootstrap 5.1.
https://getbootstrap.com/docs/5.1/forms/checks-radios/#variables

File: addons/web/static/tests/webclient/actions/client_action_tests.js [hunk 3]
```diff
@@ -258,7 +258,7 @@ QUnit.module("ActionManager", (hooks) => {
         assert.expect(1);
         class CustomControlPanel extends Component {}
         CustomControlPanel.template = xml`
-        <div class="custom-control-panel">My custom control panel</div>
+        <div class="form-check-panel">My custom control panel</div>
```
Comment (rfr-odoo) [inline addons/web/static/tests/webclient/actions/client_action_tests.js:261]: It seems not a bootstrap class

File: addons/web/static/tests/webclient/actions/client_action_tests.js [hunk 4]
```diff
@@ -270,7 +270,7 @@ QUnit.module("ActionManager", (hooks) => {
         registerCleanup(() => delete core.action_registry.map.HelloWorldTest);
         const webClient = await createWebClient({ serverData });
         await doAction(webClient, "HelloWorldTest");
-        assert.containsOnce(target, ".custom-control-panel", "should have a custom control panel");
+        assert.containsOnce(target, ".form-check-panel", "should have a custom control panel");
```
Comment (rfr-odoo) [inline addons/web/static/tests/webclient/actions/client_action_tests.js:273]: Same as before

File: addons/website_sale/static/src/scss/website_sale.scss [hunk 5]
```diff
@@ -144,7 +144,7 @@ $o-wsale-products-layout-grid-gutter-width: min($grid-gutter-width / 2, $o-wsale
         .o_categories_collapse_title, .o_products_attributes_title {
             font-size: 0.9rem;
         }
-        .custom-control.form-check {
+        .form-check.form-check {
```
Comment (rfr-odoo) [inline addons/website_sale/static/src/scss/website_sale.scss:147]: Duplicate and in this case increase the CSS specificity rule

File: addons/mail/static/src/components/composer_suggested_recipient/composer_suggested_recipient.xml [hunk 6]
```diff
@@ -3,7 +3,7 @@
     <t t-name="mail.ComposerSuggestedRecipient" owl="1">
         <t t-if="composerSuggestedRecipientView">
             <div class="o_ComposerSuggestedRecipient" t-attf-class="{{ className }}" t-att-data-partner-id="composerSuggestedRecipientView.suggestedRecipientInfo.partner and composerSuggestedRecipientView.suggestedRecipientInfo.partner.id ? composerSuggestedRecipientView.suggestedRecipientInfo.partner.id : false" t-att-title="composerSuggestedRecipientView.suggestedRecipientInfo.titleText" t-ref="root">
-                <div class="custom-control form-check">
+                <div class="form-check form-check">
```
Comment (seb-odoo) [inline addons/mail/static/src/components/composer_suggested_recipient/composer_suggested_recipient.xml:6]: duplicate class

**Comment by adr-odoo** [comment]
@robodoo delegate=rfr-odoo

**Review by rfr-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**